### DISCUSSION TITLE
Missing lifecycle of child component on parent update

### DISCUSSION THREAD

**Discussion by ju5t** [livewire/livewire#7989] [1. Help] [answered]
I'm working on the following:

- A main component.
- A child component that is used to verify the VAT ID of someone.

The child component receives the `$customer` or `$supplier` object together with the VAT ID that has been specified. In the child component, there's a reactive property like so:

```php
#[Reactive]
public ?string $vat = null;
```

This works (v3 keeps impressing me, it's incredible). I can see that the `$vat` has been updated. 

But when `$vat` is updated, it doesn't seem trigger any lifecycle hooks. This means I can only run validation in `render()` which is not my preferred solution. Am I right that lifecycle hooks are not initiated when the parent updates and if so, is there an alternative that could work here?

**Comment by danie-ramdhani** [comment, marked as answer]
maybe there's way using alpine (put this in the root element or script)?

```js
$wire.$watch('vat', (value) => {
   $wire.validateReactiveProp();
});
```

and, in child component:

```php
public function validateReactiveProp()
{
    //
}
```

or maybe check if you want to live update with extra prop in the child component:

```blade
@if ($isLive)
   $wire.$watch('vat', (value) => {
       $wire.validateReactiveProp();
   });
@endif
```

**Reply by ju5t** [reply]
Smart! That works for me. Thanks!

**Reply by whousehomestay** [reply]
This is good, but to be noticed, this code makes another request through network to execute the function.
